en/Data Structures/Arrays/array.md

```diff
@@ -0,0 +1,13 @@
+# Arrays
+
+Arrays are the most basic data structures used in programming. They contain a group of a fixed number of elements that occupy the same space in memory. I.e., the memory occupied is known at compile time. Thus arrays are stored in the stack, and their values can be saved and retrieved very quickly in `O(1)` time. Languages like `Rust` which allows `enums` in Arrays, each element occupies the maximum space which could be required by it.
```

Comment:
The memory occupied is not necessarily known at compile time. This depends on your array definition; Go calls arrays with fixed length at compile time arrays and arrays with "fixed" length at runtime slices.

Reply:
Considering that we are defining array size while initialization then memory size will be fixed. I think by stack @raklaptudirm mean some kind of pile. And the rust is just to take as an example of variant that also exist ig. 😕

Reply:
> Considering that we are defining array size while initialization then memory size will be fixed. I think by stack @raklaptudirm mean some kind of pile. And the rust is just to take as an example of variant that also exist ig. confused

Memory size will usually be fixed, yes. But note that stack != heap. Java stores objects, including arrays, on the heap.

Reply:
Yeah, stack is not used generally for dynamic size variables.